### PR TITLE
MOTECH-2028: Made unique editable from schema editor UI

### PR DESCRIPTION
**Pull request by pgesek**
* Only non DDE fields
* Only non collection types

### CODE REVIEW

**Comment by pgesek** [comment]
test this psql

**Comment by pgesek** [comment]
test this psql

**Comment by pgesek** [comment]
test this psql

**Comment by pgesek** [comment]
test this psql

File: platform/mds/mds/src/main/resources/db/migration/default/V45__MOTECH-2028.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-- add unique column in Entity table --
+
+ALTER TABLE "Field" ADD "uniqueField" boolean NOT NULL DEFAULT FALSE;
+
```
Comment (tstalka) [inline platform/mds/mds/src/main/resources/db/migration/default/V45__MOTECH-2028.sql]: add new fields to the EntityDraft

**Comment by tstalka** [comment]
What if we already have the data in the entity?

**Comment by pgesek** [comment]
If it violates the new constraint, then the schema update will fail. We can add fancy validation later on, for now this is sufficient.

**Comment by tstalka** [comment]
Add new Entity, add field , mark as unique, unmark and save changes

[ex.txt](https://github.com/motech/motech/files/66437/ex.txt)

**Comment by pgesek** [comment]
test this psql

**Comment by pgesek** [comment]
test this psql

**Comment by pgesek** [comment]
test this please

File: platform/mds/mds/src/main/java/org/motechproject/mds/builder/impl/MDSConstructorImpl.java [hunk 2]
```diff
@@ -304,6 +307,34 @@ public void updateFields(Long entityId, Map<String, String> fieldNameChanges) {
         }
     }
 
+    @Override
+    @Transactional
```
Comment (tstalka) [inline platform/mds/mds/src/main/java/org/motechproject/mds/builder/impl/MDSConstructorImpl.java]: @Transactional("dataTransactionManager")

**Pull request merged**